binding/HarfBuzzSharp.Shared/DelegateProxies.font.cs

```diff
@@ -0,0 +1,223 @@
+			var glyphs = new ReadOnlySpan<uint> (firstGlyph, (int)count);
+			var advances = new Span<int> (firstAdvance, (int)count);
+			var userData = GetFontData<FontUserData> (fontData);
+			del.Invoke (userData.Font, userData.FontData, count, glyphs, advances);
+		}
+
+		[MonoPInvokeCallback (typeof (GlyphOriginProxyDelegate))]
+		private static bool GlyphOriginProxyImplementation (IntPtr font, IntPtr fontData, uint glyph, out int x, out int y, IntPtr context)
+		{
+			var del = GetMulti<GlyphOriginDelegate> (context, out _);
+			var userData = GetFontData<FontUserData> (fontData);
+			return del.Invoke (userData.Font, userData.FontData, glyph, out x, out y);
+		}
+
+		[MonoPInvokeCallback (typeof (GlyphKerningProxyDelegate))]
+		private static int GlyphKerningProxyImplementation (IntPtr font, IntPtr fontData, uint firstGlyph, uint secondGlyph, IntPtr context)
+		{
+			var del = GetMulti<GlyphKerningDelegate> (context, out _);
+			var userData = GetFontData<FontUserData> (fontData);
+			return del.Invoke (userData.Font, userData.FontData, firstGlyph, secondGlyph);
+		}
+
+		[MonoPInvokeCallback (typeof (GlyphExtentsProxyDelegate))]
+		private static bool GlyphExtentsProxyImplementation (IntPtr font, IntPtr fontData, uint glyph, out GlyphExtents extents, IntPtr context)
+		{
+			var del = GetMulti<GlyphExtentsDelegate> (context, out _);
+			var userData = GetFontData<FontUserData> (fontData);
+			return del.Invoke (userData.Font, userData.FontData, glyph, out extents);
+		}
+
+		[MonoPInvokeCallback (typeof (GlyphContourPointProxyDelegate))]
+		private static bool GlyphContourPointProxyImplementation (IntPtr font, IntPtr fontData, uint glyph, uint pointIndex, out int x, out int y, IntPtr context)
+		{
+			var del = GetMulti<GlyphContourPointDelegate> (context, out _);
+			var userData = GetFontData<FontUserData> (fontData);
+			return del.Invoke (userData.Font, userData.FontData, glyph, pointIndex, out x, out y);
+		}
+
+		[MonoPInvokeCallback (typeof (GlyphNameProxyDelegate))]
+		private static bool GlyphNameProxyImplementation (IntPtr font, IntPtr fontData, uint glyph, char* nameBuffer, int size, IntPtr context)
+		{
+			var del = GetMulti<GlyphNameDelegate> (context, out _);
+			var userData = GetFontData<FontUserData> (fontData);
+			var name = new Span<char> (nameBuffer, size);
+			var result = del.Invoke (userData.Font, userData.FontData, glyph, out var realName);
+
+			var nameSpan = realName.AsSpan ();
+			var bufferSpan = new Span<char> (nameBuffer, size);
+			nameSpan.CopyTo (bufferSpan);
+
+			return result;
+		}
+
+		[MonoPInvokeCallback (typeof (GlyphFromNameProxyDelegate))]
+		private static bool GlyphFromNameProxyImplementation (IntPtr font, IntPtr fontData, char* name, int len, out uint glyph, IntPtr context)
+		{
+			var del = GetMulti<GlyphFromNameDelegate> (context, out _);
+			var userData = GetFontData<FontUserData> (fontData);
+
+			var actualName = len < 0
+				? new string (name)
+				: new string (name, 0, len);
```

Comment:
Create a string from the buffer, keeping in mind that a `-1` length is an null-terminated string.